base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable_obj)
+    super()
+    @nameable_obj = nameable_obj
+  end
+
+  def correct_name
+    @nameable_obj.correct_name
+  end
```

Comment:
- [ ]  your decorator class is missing` attr_accessor :nameable` kindly add it to match the requirements 🙏

Reply:
Thanks for your observation, the file has been updated.